### PR TITLE
~~add remaining methods from sharkfin.population~~ to be spread into smaller PRs

### PR DESCRIPTION
**Pull request by alanlujan91**
Incorporate methods `agent_df` and `class_stats` from sharkfin.population into macro.agent_population. 

Related to #43 and #81.

### CODE REVIEW

**Comment by sbenthall** [comment]
Thanks for this effort in the right direction, @alanlujan91 

Problematically, this PR moves functionality from `hark_agent_population.py` to `macro/agent_population.py`

This was an understandable move given that we have not very effectively communicated some of the recent refactoring, and that refactoring is not yet complete.

- We are trying to melt down `hark_agent_population.py` into smaller modules in the `sharkfin` package. The duplicate code remaining in `hark_agent_population.py` is something we want to get rid of #75 . The simulation code that needs to be modified to fix #81 is in `sharkfin.simulation`
- Likewise, the agent population code that is being used in the newer simulations is `sharkfin.population`

I'm sorry the repository is in such messy shape at the moment.

If your goal is to fix #81, then could I ask you to keep working on this PR until the changes are worked through `sharkfin.population` and `sharkfin.simulation`?

File: sharkfin/population.py [hunk 1]
```diff
@@ -97,94 +214,116 @@ def class_stats(self, store=False):
...
 
-            agent.initialize_sim()
+class AgentPopulationSolution:
+    def __init__(self, agent_population):
+        self.agent_population = agent_population
+
+        self.dist_params = self.agent_population.dist_params
+        self.database = self.agent_population.database
+
+    def merge_solutions(self, continuous_states):
```
Comment (alanlujan91) [inline sharkfin/population.py:277]: @sbenthall need review and comments for this. 

I tried to start generic, but couldn't wrap my head around it, so I did it using the framework I know to expect...

**Comment by sbenthall** [comment]
@alanlujan91 would you like me to review this PR now?

**Comment by alanlujan91** [comment]
@sbenthall I don't think that this is ready, but I would like for you to look at it and let me know if there are better ways to do this. I tried to develop a generic method to `merge` solutions, but ultimately I couldn't wrap my mind around how to do this, so I ended up just doing it based on what I understand we need for white shark, but it's not generalizable.

**Comment by alanlujan91** [comment]
@sbenthall see https://github.com/sbenthall/SHARKFin/blob/ce7d9167ba7b6de1794ea2d4004b52c9b78cad65/macro/population.ipynb

**Comment by sbenthall** [comment]
I've got a failing test because the PR does not yet add xarray as a dependency:
https://gist.github.com/sbenthall/0cc277aca319247d71f560aec6f3321e

Please add this to requirements.txt

Once I install xarray...
I've got another error due to the `time_vary` attribute.
You can reproduce these by running the existing test suite:

https://gist.github.com/sbenthall/26f5662c5e93fdf7529c976e69779ecf

The notebook looks quite interesting.
Pandas has some good built-in functionality which you might exploit for looking up the solution functions.

You could convert the dataframe to a [multiindex](https://pandas.pydata.org/docs/user_guide/advanced.html) over the ex ante heterogeneous parameters (CRRA, DiscFac) and then look up the row of the dataframe by those labels directly.
Example code:

```
> sdf = solution.merge_solutions(["RiskyAvg", "RiskyStd"])
> mi_sdf = sdf.set_index(['CRRA','DiscFac']) # creates a dataframe with a multi-index over the CRRA and DiscFac columns
> mi_sdf.loc[6.0, 0.9465000000000001]['cFunc'] # looks up the row in the 'location' in the multindex, then returns the value of the 'cFunc' column in that row
<HARK.interpolation.LinearInterpOnInterp2D at 0x7f3ff4422130>
```

**Comment by alanlujan91** [comment]
@sbenthall the test suite assumes the previous `AgentPopulation` so I have to go in and fix that. However, the test is running an actual simulation, which we still haven't pinned down how to do with the new `AgentPopulation`.

**Pull request closed without merging**